src/Agent.Plugins/TestResultParser/PipelineTestRunPublisher.cs

```diff
@@ -0,0 +1,93 @@
+                state: TestRunState.InProgress.ToString(), isAutomated: true /*, type: RunType.NoConfigRun.ToString()*/);
+            var run = await _httpClient.CreateTestRunAsync(r, _pipelineConfig.Project);
+
+            var testResults = new List<TestCaseResult>();
+
+            foreach (var passedTest in testRun.PassedTests)
+            {
+                testResults.Add(new TestCaseResult
+                {
+                    TestCaseTitle = passedTest.Name,
+                    AutomatedTestName = passedTest.Name,
+                    DurationInMs = passedTest.ExecutionTime.TotalMilliseconds,
+                    State = "Completed",
+                    AutomatedTestType = "NoConfig",
+                    Outcome = TestOutcome.Passed.ToString()
+                });
+            }
+
+            foreach (var failedTest in testRun.FailedTests)
+            {
+                testResults.Add(new TestCaseResult
+                {
+                    TestCaseTitle = failedTest.Name,
+                    AutomatedTestName = failedTest.Name,
+                    DurationInMs = failedTest.ExecutionTime.TotalMilliseconds,
+                    State = "Completed",
+                    AutomatedTestType = "NoConfig",
+                    Outcome = TestOutcome.Failed.ToString(),
+                    StackTrace = failedTest.StackTrace
+                });
+            }
+
+            foreach (var skippedTest in testRun.SkippedTests)
+            {
+                testResults.Add(new TestCaseResult
+                {
+                    TestCaseTitle = skippedTest.Name,
+                    AutomatedTestName = skippedTest.Name,
+                    DurationInMs = skippedTest.ExecutionTime.TotalMilliseconds,
+                    State = "Completed",
+                    AutomatedTestType = "NoConfig",
+                    Outcome = TestOutcome.NotExecuted.ToString()
+
+                });
+            }
+
+            await _httpClient.AddTestResultsToTestRunAsync(testResults.ToArray(), _pipelineConfig.Project, run.Id);
+
+            var runUpdateModel = new RunUpdateModel(state: TestRunState.Completed.ToString());
+            /*
+            var runUpdateModel = new RunUpdateModel(state: TestRunState.Completed.ToString())
+            {
+                RunSummary = new List<RunSummaryModel>()
+            };
+
+            runUpdateModel.RunSummary.Add(new RunSummaryModel(resultCount: testRun.TestRunSummary.TotalFailed, testOutcome: TestOutcome.Failed));
+            runUpdateModel.RunSummary.Add(new RunSummaryModel(resultCount: testRun.TestRunSummary.TotalPassed, testOutcome: TestOutcome.Passed));
+            runUpdateModel.RunSummary.Add(new RunSummaryModel(resultCount: testRun.TestRunSummary.TotalSkipped, testOutcome: TestOutcome.NotExecuted));
+            */
+
+            await _httpClient.UpdateTestRunAsync(runUpdateModel, _pipelineConfig.Project, run.Id);
```

Comment:
do we want a try catch around these client calls?

Reply:
what's the action on catching? exceptions should be propagated properly

Reply:
telemetry saying tcm publish failed

Reply:
you can rethrow after that

Reply:
Can we please have this?

Reply:
It's supposed to be handled in Test Run Manager and not in Pipeline Publisher and it's already handled there.